Sources/Build/BuildDescription/SwiftTargetBuildDescription.swift

```diff
@@ -348,7 +348,7 @@ public final class SwiftTargetBuildDescription {
             extension Foundation.Bundle {
                 static let module: Bundle = {
                     let mainPath = \(mainPathSubstitution)
-                    let buildPath = "\(bundlePath.pathString.asSwiftStringLiteralConstant)"
+                    let buildPath = "\(self.fileSystem.capitalizeDriveLetterIfOnWindows(bundlePath.pathString.asSwiftStringLiteralConstant))"
```

Comment:
Is this change related?

Reply:
cc @compnerd for opinions on this

Reply:
also this may be better expressed as an extension of AbsolutePath instead of en extension of FileSystem

Reply:
> Is this change related?

Yes, see PR description's 2nd bullet (VSCode passes "c:\..." but `swift build` gets here with "C:\..." so we need to capitalize the `c` to `C` to avoid spurious changes to the emitted files)

> also this may be better expressed as an extension of AbsolutePath instead of en extension of FileSystem

No preference on my end.

Reply:
@tomerd hmm, I think that the ideal thing would be to ensure that we always normalise the drive letter spelling.  I know that node.js will case convert to lower case, but that changes the path technically and thus causes rebuilds.  Translating to uppercase seems like the right thing to do IMO - that is the native windows spelling and the Windows APIs spell it that way.

Reply:
I agree with @compnerd, this feels like something `pathString` or `asSwiftStringLiteralConstant` should be doing vs. an extra manual step.

Reply:
I would be fine with that approach if that is the canonical representation on windows

Reply:
we should consider splitting this into 2 PRs. this one can focus on the write-if-changed, and a second one can focus on canonicalizing / normalizing the drive letter in AbsolutePath on windows